x/stakeibc/keeper/unbonding_records.go

```diff
@@ -134,14 +143,33 @@ func (k Keeper) GetValidatorUnbondCapacity(
 // This will also sort such that 0-weight validator's will come first as their
 //   ideal balanced delegation will always be 0, and thus their ratio will always be 0
 // If the ratio's are equal, the validator with the larger delegation/capacity will come first
-func SortUnbondingCapacityByPriority(validatorUnbondCapacity []ValidatorUnbondCapacity) []ValidatorUnbondCapacity {
+func SortUnbondingCapacityByPriority(validatorUnbondCapacity []ValidatorUnbondCapacity) ([]ValidatorUnbondCapacity, error) {
+	// Compare function has to have signature "func(i,j int) bool"
+	// If errors occur while comparing, panic, and this deferred function will catch it and update the closured error variable
+	var sortingErr error = nil
```

Comment:
I see why you were saying this was so tricky! I like the creativity, but we should avoid using the defer/recover. Instead we can just loop each val first here
```go
for _, validator := range validatorUnbondCapacity {
   if _, err := validator.GetBalanceRatio(); err != nil {
       return err
   }
}
```

then in the sort function, you can just ignore the error